src/raster/rasterband.rs

```diff
@@ -556,6 +562,41 @@ impl<'a> RasterBand<'a> {
         Array2::from_shape_vec((size.1, size.0), data).map_err(Into::into)
     }
 
+    #[cfg(feature = "ndarray")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "array")))]
+    /// Write a [`Array2<T>`] from a [`Dataset`] block, where `T` implements [`GdalType`].
+    ///
+    /// # Arguments
+    /// * `block_index` - the block index
+    ///
+    /// # Notes
+    /// The Matrix shape is (rows, cols) and raster shape is (cols in x-axis, rows in y-axis).
+    pub fn write_block<T: Copy + GdalType>(
+        &mut self,
+        block_index: (usize, usize),
```

Comment:
What happens if the block index is out of bounds? Do we get an error, or `panic` (or worse)?

Reply:
There's no way it can panic, of course :-). Added a note in the docs.